### ISSUE TITLE
Change to generic with labelContent

### ISSUE THREAD

**Issue by shahankit-spotnana** [googlemaps/js-markerwithlabel#409]
#### Environment details

1. Specify the API at the beginning of the title (for example, "Places: ...")
  - MarkerWithLabel.labelContent 
2. OS type and version
  - OSX
3. Library version and other environment information
  - version: 1.3.10

#### Steps to reproduce

I'm trying to access marker's label node, but when I try to get it via `marker.labelContent`, it returns the innerHTML representation. Seem like [code](https://github.com/googlemaps/js-markerwithlabel/blob/main/src/label.ts#L83) for getter has return type as `string | HTMLElement` but we are always returning innerHTML string. Is this expected or a miss in implementation?

Reason for access to labelContent: I'm using React.createPortal to render children to marker.labelContent node.

#### Rough demo code

```tsx
function CustomMarker({ children }) {
  const [mapMarker, setMapMarker] = useState<MarkerWithLabel | null>(null);

  useEffect(() => {
    const labelContentEl = document.createElement('div');
    const marker = new MarkerWithLabel({ labelContent: labelContentEl });
    setMapMarker(marker);
  }, []);

  // This fails because mapMarker.labelContent is string.
  return mapMarker ? React.createPortal(children, mapMarker.labelContent) : null;
}
```

**Comment by jpoehnelt** [comment]
There are two ways forward:

1. Use a TypeScript guard to narrow the union of `string | HTMLElement`... `labelContent instanceof HTMLScriptElement`.
2. Modify the library to be MarkerLabel<HTMLElement> and use the generic form for get/set on label content.

**Comment by jpoehnelt** [comment]
Actually you are correct in that innerHTML only returns the string. Ignore the TS discussion above :)

**Comment by shahankit-spotnana** [comment]
@jpoehnelt Thanks a lot for quick response and fix.

**Comment by jpoehnelt** [comment]
👍 

Heads up that I also published v2.0.0 today which switch to named exports #411.

**Comment by shahankit-spotnana** [comment]
Yes, I've upgraded to v2.0.0.
